GVFS/GitHooksLoader/GitHooksLoader.cpp

```diff
@@ -112,9 +112,6 @@ int ExecuteHook(const std::wstring &applicationName, wchar_t *hookName, int argc
     PROCESS_INFORMATION pi;
     ZeroMemory(&si, sizeof(si));
     si.cb = sizeof(si);
-    si.hStdOutput = GetStdHandle(STD_OUTPUT_HANDLE);
-    si.hStdError = GetStdHandle(STD_ERROR_HANDLE);
-    si.dwFlags = STARTF_USESTDHANDLES;

```

Comment:
So what you're saying is that this explicit redirection of `stdout`/`stderr` currently does _not_ work? This is puzzling because the `GitHooksLoader.exe` process _should_ have the correct handles, and this way of creating the child process _should_ work...

Reply:
Yes, it does not work. For example running git status on an unmounted gvfs repo displays only the standard "pre command hook failed" message, and not the specific message about needing to mount.

I wonder if it's somehow environment dependent; I have a vague memory of seeing that message years ago, but there has been no change to this file.

However the [documentation](https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags) for CREATE_NO_WINDOW says `The process is a console application that is being run without a console window. Therefore, the console handle for the application is not set.`

Reply:
I have updated this to more closely match how [git-wrapper in microsoft/git ](https://github.com/git-for-windows/MINGW-packages/blob/994073c2964b48d9dfcf27124160cf0b237d0239/mingw-w64-git/git-wrapper.c#L882) works - checking explicitly for a console with CONOUT$, and if found then use default behavior (ie don't set CREATE_NO_WINDOW, don't explicitly forward handles as they are forwarded by default), falling back to the current behavior if CONOUT$ is not found (set CREATE_NO_WINDOW and explicitly forward handles)